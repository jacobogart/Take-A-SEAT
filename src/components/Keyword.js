import React, { Component } from 'react';
import '../css/Keyword.scss';

class Keyword extends Component {
  render() {
    return (

    );
  }
}

export default Keyword;
