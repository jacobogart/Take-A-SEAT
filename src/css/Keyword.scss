.keywordButton {
  background-image: url('../images/pencil-button.png');
  background-size: cover;
  font-family: 'Roboto Condensed', sans-serif;
  background-position: center;
  color: #157d07;
  background-color: transparent;
  width: 240px;
  border: white;
  margin: 20px;
  &:hover {
    transform: scale(1.5);
    cursor: pointer;
  }
}