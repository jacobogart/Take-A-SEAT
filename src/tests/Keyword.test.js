import React from "react";
import Keyword from "../components/Keyword";
import { shallow } from "enzyme";

const mockUpdateCurrentWord = jest.fn();
const mockChalkboardChecker = jest.fn();
const mockKeywordData = {
  "word": "wrapper",
  "nextWords": [
    "=",
    ";",
    ")",
    "state("
  ],
  "value": "wrapper",
  "isEditable": false,
  "phase": ["S", "A"],
  "details": "An object that surrounds the shallow rendering of an instance",
  "id": 110
};

describe('Keyword', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Keyword
        key={mockKeywordData.id}
        keywordData={mockKeywordData}
        updateCurrentWord={mockUpdateCurrentWord}
        chalkboardChecker={mockChalkboardChecker}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state("showDetails")).toEqual(false);
  });

  it('should render keywordInfo if showDetails is true', () => {
    wrapper.instance().setState({ showDetails: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke selectWord when .keywordButton is clicked', () => {
    wrapper.find('.keywordButton').simulate('click');
    expect(wrapper.state('showDetails')).toEqual(true);
  });

  describe('selectWord', () => {
    const mockValue = "const";
    it('should toggle showDetails when selectWord is called', () => {
      expect(wrapper.state('showDetails')).toEqual(false);
      wrapper.instance().selectWord(mockValue);
      expect(wrapper.state('showDetails')).toEqual(true);
      wrapper.instance().selectWord(mockValue);
      expect(wrapper.state('showDetails')).toEqual(false);
    })

    it('should invoke updateCurrentWord if showDetails is true', () => {
      wrapper.instance().setState({ showDetails: true});
      wrapper.instance().selectWord(mockValue);
      expect(mockUpdateCurrentWord).toHaveBeenCalled();
      expect(mockUpdateCurrentWord).toHaveBeenCalledWith(mockKeywordData);
    });

    it('should invoke chalkboardChecker if showDetails is true', () => {
      wrapper.instance().setState({ showDetails: true });
      wrapper.instance().selectWord(mockValue);
      expect(mockChalkboardChecker).toHaveBeenCalled();
      expect(mockChalkboardChecker).toHaveBeenCalledWith('const');
    });
  })
})
