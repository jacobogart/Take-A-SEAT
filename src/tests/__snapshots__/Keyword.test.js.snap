// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keyword should match snapshot 1`] = `
<div>
  <button
    className="keywordButton"
    onClick={[Function]}
  >
    wrapper
  </button>
</div>
`;

exports[`Keyword should render keywordInfo if showDetails is true 1`] = `
<div>
  <button
    className="keywordButton"
    onClick={[Function]}
  >
    wrapper
  </button>
  <KeywordInfo
    keywordData={
      Object {
        "details": "An object that surrounds the shallow rendering of an instance",
        "id": 110,
        "isEditable": false,
        "nextWords": Array [
          "=",
          ";",
          ")",
          "state(",
        ],
        "phase": Array [
          "S",
          "A",
        ],
        "value": "wrapper",
        "word": "wrapper",
      }
    }
    selectWord={[Function]}
  />
</div>
`;
